File: src/main/java/sodium/StreamSink.java
package sodium;

public class StreamSink<A> extends StreamWithSend<A> {
    public StreamSink() {}

	public void send(final A a) {
		Transaction.Companion.run(new Handler<Transaction>() {
            public void run(Transaction trans) {
                if (Transaction.inCallback > 0)
                    throw new RuntimeException("You are not allowed to use send() inside a Sodium callback");
                send(trans, a);
            }
        });
	}
}

File: src/main/kotlin/sodium/StreamSink.kt
package sodium

public class StreamSink<A> : sodium.impl.StreamWithSend<A>() {

    public fun send(a: A) {
        Transaction.apply2 {
            if (Transaction.inCallback > 0)
                throw RuntimeException("You are not allowed to use send() inside a Sodium callback")
            send(it, a)
        }
    }
}
